### PR TITLE
Update `BATSModel`, `TBATSModel` models to work with integer timestamp

### LINKED ISSUE #181: Update `BATSModel`, `TBATSModel` to handle integer timestamp
### 🚀 Feature Request

In #178 it was found that tbats models fail on data with integer timestamp. It could be fixed.

### Proposal

Fix tests `tests/test_models/test_inference/test_forecast/TestForecastOutSample` for models:
- `BATSModel`
- `TBATSModel`

Error:
```
ValueError: Model is not fitted! Fit the model before calling predict method!
```

The problem is with `_TBATSAdapter.forecast` which checks `if self._freq is None` which is now a valid value. 
- The default value for `self._freq` should be changed.
- The check on fitted model should be changed.

### Test cases

- Move fixed tests in `tests/test_models/test_inference/test_forecast/TestForecastOutSample` from fail to success category
- Move fixed tests `tests/test_models/test_inference/test_predict/TestPredictInSampleSuffix` from fail to success category

### Additional context

Blocked by #178.

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #181 , additionaly typing for `SARIMAXModel`, `AutoARIMAModel` was fixed.

## Closing issues
Closes #181.

### CODE REVIEW

File: tests/test_models/test_tbats.py [hunk 1]
```diff
@@ -375,6 +394,7 @@ def test_arma_with_seasonal_components_not_fitted(periodic_dfs, estimator, metho
 
 
 @pytest.mark.filterwarnings("ignore:.*not fitted.*")
+@pytest.mark.parametrize("dfs_name", ["periodic_dfs", "periodic_dfs_int_timestamp"])
```
Comment (d-a-bunin) [inline tests/test_models/test_tbats.py:397]: I'm not sure it is a good place to check components for int timestamp.

**Comment by ostreech1997** [comment]
Why do we need `regressors` argument here? I see that we don't use it.

https://github.com/etna-team/etna/blob/e016cb4179d051ca239504fbf236646ac54f760e/etna/models/tbats.py#L43

**Comment by d-a-bunin** [comment]
You are correct, we aren't using `regressors` here. It is only needed for compatibility reasons. Model implements unspecified interface which requires the parameter `regressors` to exist.

File: tests/test_models/test_tbats.py (hunk 1 above)
Comment (d-a-bunin) [reply tests/test_models/test_tbats.py:397]: Here I check that components are built correctly if timestamp is integer. But I'm not sure that adding df with integer timestamp in exactly this test is a good idea. We could probably add this case into some other test that works with components prediction.

Comment (ostreech1997) [reply tests/test_models/test_tbats.py:397]: I think, It's okey. What do you see as the problem?

**Review by ostreech1997 — APPROVED** [review]

**Pull request merged**